File: Clg_Placement/C++/_41.QuickLeftRotation.cpp
class Solution{
	public:
	void leftRotate(int arr[], int k, int n) 
	{ 
	   k = k%n;   //Never forget to code this step k = k%n;
	   reverse(arr,arr+k);
	   reverse(arr+k,arr+n);
	   reverse(arr,arr+n);
	} 
};

File: Clg_Placement/Java/_41.QuickLeftRotation.java
class Solution
{
    void reverse(long arr[],int start,int end){
        while(start<end){
            arr[start] = arr[start]^arr[end];
            arr[end] = arr[start]^arr[end];
            arr[start] = arr[start]^arr[end];
            start++;
            end--;
        }
    }
    void leftRotate(long arr[], int k,int n)
    {
        k = k%n;
        reverse(arr,0,k-1);
        reverse(arr,k,n-1);
        reverse(arr,0,n-1);
    }
}
